packages/core/src/render3/interfaces/definition.ts

```diff
@@ -241,6 +243,9 @@ export interface ComponentDef<T> extends DirectiveDef<T> {
    */
   readonly template: ComponentTemplate<T>;
 
+  /** Constants associated with the component's view. */
+  readonly consts: TAttributes[]|null;
```

Comment:
For follow up can we change this to:
```suggestion
  readonly consts: () => TAttributes[]|null;
```